compression/cutils/tests/utils/compression.ts

```diff
@@ -0,0 +1,382 @@
+            updateAuthority: payer.publicKey,
+            metadata: metadataAccount,
+        },
+        {
+            createMasterEditionArgs: {
+                maxSupply: 0,
+            },
+        },
+    );
+
+    // create the collection size instruction
+    const collectionSizeIX = createSetCollectionSizeInstruction(
+        {
+            collectionMetadata: metadataAccount,
+            collectionAuthority: payer.publicKey,
+            collectionMint: mint,
+        },
+        {
+            setCollectionSizeArgs: { size: 50 },
+        },
+    );
+
+    try {
+        // construct the transaction with our instructions, making the `payer` the `feePayer`
+        const tx = new Transaction()
+            .add(createMetadataIx)
+            .add(createMasterEditionIx)
+            .add(collectionSizeIX);
+        tx.feePayer = payer.publicKey;
+
+        // send the transaction to the cluster
+        const txSignature = await sendAndConfirmTransaction(connection, tx, [payer], {
+            commitment: "confirmed",
+            skipPreflight: true,
+        });
+
+        console.log("\nCollection successfully created!");
+        console.log(explorerURL({ txSignature }));
+    } catch (err) {
+        console.error("\nFailed to create collection:", err);
+
+        // log a block explorer link for the failed transaction
+        await extractSignatureFromFailedTransaction(connection, err);
+
+        throw err;
+    }
+
+    // return all the accounts
+    return { mint, tokenAccount, metadataAccount, masterEditionAccount };
+}
+
+/**
+ * Mint a single compressed NFTs to any address
+ */
+export async function mintCompressedNFT(
+    connection: Connection,
+    payer: Keypair,
+    treeAddress: PublicKey,
+    collectionMint: PublicKey,
+    collectionMetadata: PublicKey,
+    collectionMasterEditionAccount: PublicKey,
+    compressedNFTMetadata: MetadataArgs,
+    receiverAddress?: PublicKey,
+) {
+    // derive the tree's authority (PDA), owned by Bubblegum
+    const [treeAuthority, _bump] = PublicKey.findProgramAddressSync(
+        [treeAddress.toBuffer()],
+        BUBBLEGUM_PROGRAM_ID,
+    );
+
+    // derive a PDA (owned by Bubblegum) to act as the signer of the compressed minting
+    const [bubblegumSigner, _bump2] = PublicKey.findProgramAddressSync(
+        // `collection_cpi` is a custom prefix required by the Bubblegum program
+        [Buffer.from("collection_cpi", "utf8")],
+        BUBBLEGUM_PROGRAM_ID,
+    );
+
+    // create an array of instruction, to mint multiple compressed NFTs at once
+    const mintIxs: TransactionInstruction[] = [];
+
+    /**
+     * correctly format the metadata args for the nft to mint
+     * ---
+     * note: minting an nft into a collection (via `createMintToCollectionV1Instruction`)
+     * will auto verify the collection. But, the `collection.verified` value inside the
+     * `metadataArgs` must be set to `false` in order for the instruction to succeed
```

Comment:
Is this a bug we should report to the team? What is the resulting error?

Reply:
hmm good point, maybe this should be documented somewhere else.

At least would be good to somehow inform the user on this when using the metaplex SDK? What does @nickfrosty think?